rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/UserService.java

```diff
@@ -0,0 +1,80 @@
+package org.niewidoczniakademicy.rezerwacje.service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.validator.routines.EmailValidator;
+import org.niewidoczniakademicy.rezerwacje.core.model.database.SystemUser;
+import org.niewidoczniakademicy.rezerwacje.core.model.rest.user.AddSystemUserRequest;
+import org.niewidoczniakademicy.rezerwacje.core.model.rest.user.AddSystemUserResponse;
+import org.niewidoczniakademicy.rezerwacje.core.model.rest.user.GetSystemUserResponse;
+import org.niewidoczniakademicy.rezerwacje.core.model.rest.user.GetSystemUsersResponse;
+import org.niewidoczniakademicy.rezerwacje.dao.repository.UserRepository;
+import org.niewidoczniakademicy.rezerwacje.service.converter.ConversionService;
+import org.niewidoczniakademicy.rezerwacje.service.exception.InvalidEmailAddressException;
+import org.niewidoczniakademicy.rezerwacje.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Slf4j
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class UserService {
+
+    private final ConversionService conversionService;
+    private final UserRepository userRepository;
+
+    public AddSystemUserResponse saveSystemUser(AddSystemUserRequest request) {
+        validateAddSystemUserRequest(request);
+
+        SystemUser systemUser = conversionService.convert(request);
+        userRepository.save(systemUser);
+
+        return AddSystemUserResponse.builder()
+                .userId(systemUser.getId())
+                .build();
+    }
+
+    public GetSystemUserResponse getSystemUserByUserUniqueId(String userUniqueId) {
+        final SystemUser systemUser = userRepository.findByUserUniqueId(userUniqueId);
```

Comment:
Chyba lepiej, żeby dao zwracało optionala i byśmy mogli ominąć ify i użyć orElseThrow, ewentualnie Optional.ofNullable do pozbycia się tych ifów? W sumie to kwestia kosmetyczna, mogą zostać te ify.

Reply:
Ok poprawie